platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/ImportMigrationContext.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+/**
+ * The import migration context keeps track of exported migration entries for a given migratable
+ * while processing an import migration operation.
+ * <p>
+ * <b>
+ * This code is experimental. While this interface is functional
+ * and tested, it may change or be removed in a future version of the
+ * library.
+ * </b>
+ * </p>
+ */
+public interface ImportMigrationContext extends MigrationContext {
+    /**
+     * Gets the exported version for the corresponding migratable.
+     *
+     * @return the exported version for the corresponding migratable or <code>null</code> if the
+     * corresponding migratable was not exported
+     */
+    @Nullable
+    public String getVersion();
```

Comment:
It wasn't originally for 2 reasons:
1- we didn't enforce it for a Migratable
2- we had to account for the case where a Migratable is now installed but was never exported (in upgrade scenarios)

After some discussions, I have changed that to make the version on the migratable required although the format is not enforced and is migratable-dependent as long as it is not null. The infrastructure will only compare them using equals(). If the migratable was not exported, a new Migratable.doMissingImport() will be called. If the version exported is equal to the migratable current version, Migratable.doImport() will be called and if it is not equal, Migratable.doIncompatibleImport() will be called.